.github/workflows/github-cxx-qt-tests.yml

```diff
@@ -88,6 +88,134 @@ jobs:
           token: ${{ secrets.CODECOV_TOKEN }}
           verbose: true
 
+  build-wasm:
+    name: Ubuntu 22.04 (wasm_32) Qt 6
+    needs: [clang_format, license_check, rust_format_check, markdown_lint, shellcheck]
+    runs-on: ubuntu-22.04
+    env:
+      SCCACHE_CACHE_SIZE: 600M
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+      - name: Setup toolchain
+        run: |
+          rustup component add clippy rustfmt
+          rustup target add wasm32-unknown-emscripten
+      - name: "Rust tools cache"
+        uses: actions/cache@v4
+        id: rust-tools-cache
+        with:
+          path: |
+            ~/.cargo/bin/sccache
+            ~/.cargo/bin/mdbook
+            ~/.cargo/bin/mdbook-linkcheck
+          key: ubuntu-22.04_sccache-0.7.6_mdbook-0.4.36_mdbook-linkcheck-0.7.7
+      - name: "Build Rust tools"
+        if: steps.rust-tools-cache.outputs.cache-hit != 'true'
+        # Do not build with storage backends enabled, we only need local
+        run: |
+          cargo install --no-default-features sccache
+          cargo install mdbook mdbook-linkcheck
+      # We want our compiler cache to always update to the newest state.
+      # The best way for us to achieve this is to **always** update the cache after every landed commit.
+      # That way it will closely follow our development.
+      # And if a PR diverges a lot with its cache that's not a big deal, as it will be merged eventually.
+      #
+      # This is a workaround for the fact that GH doesn't support updating existing caches.
+      # See: https://github.com/azu/github-actions-overwrite-cache-example
+      #
+      # Ideally we'd like to use this:
+      # - name: "Compiler cache"
+      #   uses: actions/cache@v4
+      #   with:
+      #     update: true <------- THIS DOESN'T EXIST YET
+      #     path: /home/runner/.cache/sccache
+      #     key: "Ubuntu 22.04 (wasm_32) Qt6_compiler_cache"
+      - name: "Restore Compiler Cache"
+        id: compiler-cache-restore
+        uses: actions/cache/restore@v4
+        with:
+          path: /home/runner/.cache/sccache
+          key: "Ubuntu 22.04 (wasm_32) Qt6_compiler_cache"
+      - name: "Setup emsdk"
+        run: |
+          git clone https://github.com/emscripten-core/emsdk
+          ./emsdk/emsdk install 2.0.14
+          ./emsdk/emsdk activate 2.0.14
+      - name: "emsdk cache"
+        uses: actions/cache@v4
+        id: emsdk-cache
+        with:
+          path: ./emsdk
```

Comment:
You only need to setup if there is no cache, so swap them around and add a condition. Also we need the cache id to be versioned otherwise we cannot bump the emsdk version later to break out of the cache.

So it should look like this in the end

```
      - name: "emsdk cache"
        uses: actions/cache@v4
        id: emsdk-cache-2.0.14
        with:
          path: ./emsdk
      - name: "Setup emsdk"
        if: steps.emsdk-cache.outputs.cache-hit != 'true'
        run: |
          git clone https://github.com/emscripten-core/emsdk
          ./emsdk/emsdk install 2.0.14
          ./emsdk/emsdk activate 2.0.14
```

I assume that the `install` command only puts things inside the `emsdk` folder and not in some node_modules folder elsewhere?

Reply:
Ah wait i missed the `key` somehow in my copy, so the version is there already!

So just the reorder and if statement
```
      - name: "emsdk cache"
        uses: actions/cache@v4
        id: emsdk-cache
        with:
          path: ./emsdk
          key: emsdk_2.0.14
      - name: "Setup emsdk"
        if: steps.emsdk-cache.outputs.cache-hit != 'true'
        run: |
          git clone https://github.com/emscripten-core/emsdk
          ./emsdk/emsdk install 2.0.14
          ./emsdk/emsdk activate 2.0.14
```

Reply:
i believe everything is within `emsdk`:

```console
╔ 📁 emsdk   main  🐍 v3.10.12 
╚══ ls -l --group-directories-first
total 256
drwxrwxr-x 6 sus sus   4096 Sep 12 10:54 bazel
drwxrwxr-x 2 sus sus   4096 Sep 12 10:54 docker
drwxrwxr-x 2 sus sus   4096 Sep 16 11:38 downloads
drwxrwxr-x 3 sus sus   4096 Sep 12 10:56 node
drwxrwxr-x 2 sus sus   4096 Sep 12 10:54 scripts
drwxrwxr-x 2 sus sus   4096 Sep 12 10:54 test
drwxrwxr-x 6 sus sus   4096 Sep 16 11:41 upstream
```

Reply:
^ there are other things in the dir too but those are the subfolders which contain for example a node distro etc

Reply:
Right lets go with it for now :-)